components/org.wso2.micro.integrator.coordination/src/main/java/org/wso2/micro/integrator/coordination/query/QueryManager.java

```diff
@@ -0,0 +1,303 @@
+
+                put(DBQueries.GET_COORDINATOR_NODE_ID, "SELECT NODE_ID FROM LEADER_STATUS_TABLE WHERE GROUP_ID =?");
+                put(DBQueries.GET_COORDINATOR_NODE, "SELECT PROPERTY_MAP FROM CLUSTER_NODE_STATUS_TABLE WHERE GROUP_ID =? AND NODE_ID =?");
+                put(DBQueries.GET_COORDINATOR_ROW_FOR_NODE_ID, "SELECT LAST_HEARTBEAT FROM LEADER_STATUS_TABLE WHERE NODE_ID =? AND GROUP_ID =?");
+                put(DBQueries.GET_COORDINATOR_HEARTBEAT, "SELECT LAST_HEARTBEAT FROM LEADER_STATUS_TABLE WHERE GROUP_ID =?");
+                put(DBQueries.GET_ALL_NODE_HEARTBEAT, "SELECT GROUP_ID, NODE_ID, PROPERTY_MAP, LAST_HEARTBEAT, IS_NEW_NODE FROM CLUSTER_NODE_STATUS_TABLE WHERE GROUP_ID=?");
+                put(DBQueries.GET_NODE_DATA, "SELECT GROUP_ID, NODE_ID, PROPERTY_MAP, LAST_HEARTBEAT, IS_NEW_NODE FROM CLUSTER_NODE_STATUS_TABLE WHERE GROUP_ID=? AND NODE_ID=?");
+
+                put(DBQueries.DELETE_COORDINATOR, "DELETE FROM LEADER_STATUS_TABLE WHERE GROUP_ID =? AND LAST_HEARTBEAT < ?");
+                put(DBQueries.DELETE_NODE_HEARTBEAT, "DELETE FROM CLUSTER_NODE_STATUS_TABLE WHERE NODE_ID =? AND GROUP_ID =?");
+                put(DBQueries.DELETE_REMOVED_MEMBER_DETAIL_FOR_NODE, "DELETE FROM REMOVED_MEMBERS_TABLE WHERE NODE_ID =? AND REMOVED_MEMBER_ID =? AND  GROUP_ID =?");
+                put(DBQueries.CLEAR_NODE_HEARTBEATS, "DELETE FROM CLUSTER_NODE_STATUS_TABLE");
+                put(DBQueries.CLEAR_COORDINATOR_HEARTBEAT, "DELETE FROM LEADER_STATUS_TABLE");
+                put(DBQueries.CLEAR_ALL_MEMBERSHIP_EVENTS, "DELETE FROM  MEMBERSHIP_EVENT_TABLE");
+                put(DBQueries.CLEAN_MEMBERSHIP_EVENTS_FOR_NODE, "DELETE FROM MEMBERSHIP_EVENT_TABLE WHERE NODE_ID =?");
+            }});
+
+    public enum DBQueries {
+        IS_LEADER_STATUS_TABLE_EXISTS,
+        IS_CLUSTER_NODE_STATUS_TABLE_EXISTS,
+        IS_MEMBERSHIP_EVENT_TABLE_EXISTS,
+        IS_REMOVED_MEMBERS_TABLE_EXISTS,
+        CREATE_LEADER_STATUS_TABLE,
+        CREATE_CLUSTER_NODE_STATUS_TABLE,
+        CREATE_MEMBERSHIP_EVENT_TABLE,
+        CREATE_REMOVED_MEMBERS_TABLE,
+        INSERT_COORDINATOR_ROW,
+        INSERT_NODE_HEARTBEAT_ROW,
+        INSERT_MEMBERSHIP_EVENT,
+        INSERT_REMOVED_MEMBER_DETAILS,
+        UPDATE_PROPERTIES_MAP,
+        UPDATE_COORDINATOR_HEARTBEAT,
+        UPDATE_NODE_HEARTBEAT,
+        MARK_NODE_NOT_NEW,
+        SELECT_MEMBERSHIP_EVENT,
+        SELECT_REMOVED_MEMBER_DETAILS,
+        GET_COORDINATOR_NODE_ID,
+        GET_COORDINATOR_NODE,
+        GET_COORDINATOR_ROW_FOR_NODE_ID,
+        GET_COORDINATOR_HEARTBEAT,
+        GET_ALL_NODE_HEARTBEAT,
+        GET_NODE_DATA,
+        DELETE_COORDINATOR,
+        DELETE_NODE_HEARTBEAT,
+        DELETE_REMOVED_MEMBER_DETAIL_FOR_NODE,
+        CLEAR_NODE_HEARTBEATS,
+        CLEAR_COORDINATOR_HEARTBEAT,
+        CLEAR_ALL_MEMBERSHIP_EVENTS,
+        CLEAN_MEMBERSHIP_EVENTS_FOR_NODE,
+    }
+
+    public enum DBTypes {
```

Comment:
What are the DBs we checked this with ?

Reply:
Could you commit the scripts to be executed against the DB

Reply:
We need to have the Db scripts ?

Reply:
I have tested this with MySQL only.